plugins/authentication/ldap/ldap.php

```diff
@@ -177,8 +177,8 @@ public function onUserAuthenticate($credentials, $options, &$response)
 		}
 
 		// Grab some details from LDAP and return them
-		$response->username = $entry->getAttribute($ldap_uid);
-		$response->email    = $entry->getAttribute($ldap_email);
+		$response->username = $entry->getAttribute($ldap_uid)[0] ?? false;
+		$response->email    = $entry->getAttribute($ldap_email)[0] ?? false;
 		$response->fullname = $entry->getAttribute($ldap_fullname) ?: $credentials['username'];
```

Comment:
```suggestion
		$response->fullname = $entry->getAttribute($ldap_fullname)[0] ?? $credentials['username'];
```

Or maybe better some additional test so that `$credentials['username']` is used when `$entry->getAttribute($ldap_fullname)[0]` is an empty string? Or even when `trim($entry->getAttribute($ldap_fullname)[0])` is an empty string?

But so or so you have to use an array key also here.